frontend/src/components/editor/file-tree/upload.tsx

```diff
@@ -58,3 +68,34 @@ export function useFileExplorerUpload(options: DropzoneOptions = {}) {
     ...options,
   });
 }
+
+/**
+ * Get the path of a file.
+ *
+ * Types only have `webkitRelativePath`, but File objects in the browser
+ * can have `path` and `relativePath`.
+ */
+function getPath(file: File): string | undefined {
+  if (file.webkitRelativePath) {
+    return file.webkitRelativePath;
+  }
+  if ("path" in file && typeof file.path === "string") {
+    return file.path;
+  }
+  if ("relativePath" in file && typeof file.relativePath === "string") {
+    return file.relativePath;
+  }
+  return undefined;
+}
```

Comment:
Think we can safely update the signature here, since this utility only can return valid `FilePath`s. That way it's kind of the only "safe" way to get a our branded type, without making one our own.


```suggestion
function getPath(file: File): FilePath | undefined {
  if (file.webkitRelativePath) {
    return file.webkitRelativePath as FilePath;
  }
  if ("path" in file && typeof file.path === "string") {
    return file.path as FilePath;
  }
  if ("relativePath" in file && typeof file.relativePath === "string") {
    return file.relativePath as FilePath;
  }
  return undefined;
}
```